extension/src/pipeline/quickPick.ts

```diff
@@ -52,6 +52,69 @@ const formatFieldQuickPickValues = (
   return formattedFields
 }
 
+const verifyFilesHavingSingleField = (files: {
+  [file: string]: string[] | string
+}) => {
+  for (const [file, fields] of Object.entries(files)) {
+    if (Array.isArray(fields)) {
+      void Toast.showError(
+        `${file} cannot have more than one metric selected. See [an example](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#available-configuration-fields) of a plot with multiple x metrics.`
+      )
+      return
+    }
+  }
+
+  return files
+}
+
+const verifyXFields = (
+  xValues: { file: string; key: string }[] | undefined
+) => {
+  if (!xValues || xValues.length === 0) {
+    return
+  }
+
+  const x = formatFieldQuickPickValues(xValues)
+
+  const doFilesHaveOneField = verifyFilesHavingSingleField(x)
+
+  if (!doFilesHaveOneField) {
+    return
+  }
+
+  return { firstXKey: xValues[0].key, x: x as PlotConfigData['x'] }
+}
+
+const verifyYFields = (
+  yValues: { file: string; key: string }[] | undefined,
+  xFieldsLength: number
+) => {
+  if (!yValues || yValues.length === 0) {
+    return
+  }
+
+  const y = formatFieldQuickPickValues(yValues)
+
+  if (xFieldsLength === 1) {
+    return { firstYKey: yValues[0].key, y }
+  }
+
+  if (yValues.length !== xFieldsLength) {
+    void Toast.showError(
+      'The amount of y metrics needs to match the amount of x metrics. See [an example](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#available-configuration-fields) of a plot with multiple x metrics.'
+    )
+    return
+  }
+
+  const doFilesHaveOneField = verifyFilesHavingSingleField(y)
+
+  if (!doFilesHaveOneField) {
```

Comment:
[B] Whilst I believe this is correct for x values this is not correct for ys. This is a valid plot:

```
  - Loss:
      x: step
      y:
        training/plots/metrics/train/loss.tsv: loss
        training/plots/metrics/test/loss.tsv: loss
        training/plots/metrics/test/acc.tsv: [acc, step]
      y_label: loss
```

Reply:
so is this 
```
  - Loss:
      x: step
      y:
        training/plots/metrics/train/acc.tsv: [acc, step]
```

Reply:
> [B] Whilst I believe this is correct for x values this is not correct for ys. 

We only check for this when there are multiple `x` values. When we have an x dict, the `y` and `x` need to match format-wise.

Reply:
> When we have an x dict, the y and x need to match format-wise. I'll adjust this code to make this more clear :)

Actually, after testing `x dict`s further, this doesn't appear to be true 🤦‍♀️... I'm getting mixed results when I try to test out `x dict` plots and I'm no longer sure what we actually need to check for 😅. I'll just leave in a check for the same amount of fields and followup once I understand plots with multiple x fields better.

Reply:
Discussed with Matt synchronously, we're going to keep the extra check for y files having a single metric when there are multiple x metrics.